From f34bc58f7258d5dc3a9df2862102716414844bd5 Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Tue, 31 Jul 2012 00:06:40 +0800
Subject: [PORT FROM MAIN]fix crash issue while playing certain streaming media

BZ:49305

In the play response of certain streaming media, the url just contains part of the full url.
Change the url matching method to enable this case rather than crash as before.

orignal-patch
30070:libstagefright: patch porting from R2 to R3 for RTSP
16854[R1]:fix crash issue while playing certain streaming media(BZ:7540)

Change-Id: I69b4e74d415cdf006ae87a7e6138264af2603771
Orig-Change-Id: Idec1bf63bbc14c161fb3684fd892ad9baf36e08b
Signed-off-by: ywan171 <yi.a.wang@intel.com>
---
 media/libstagefright/rtsp/MyHandler.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/rtsp/MyHandler.h b/media/libstagefright/rtsp/MyHandler.h
index cf82688..e3ce818 100644
--- a/media/libstagefright/rtsp/MyHandler.h
+++ b/media/libstagefright/rtsp/MyHandler.h
@@ -1409,7 +1409,7 @@ struct MyHandler : public AHandler {
 
             size_t trackIndex = 0;
             while (trackIndex < mTracks.size()
-                    && !(val == mTracks.editItemAt(trackIndex).mURL)) {
+                    && (-1 == mTracks.editItemAt(trackIndex).mURL.find(val.c_str(), 0))) {
                 ++trackIndex;
             }
             CHECK_LT(trackIndex, mTracks.size());
-- 
1.7.4.1

